lib/ex_unit/lib/ex_unit/cli_formatter.ex

```diff
@@ -210,14 +210,22 @@ defmodule ExUnit.CLIFormatter do
     do: colorize([:bright, :black], msg, config)
 
   defp formatter(:diff_delete, msg, config),
-    do: colorize(:red, msg, config)
+    do: colorize(:red, show_whitespace(msg), config)
 
   defp formatter(:diff_insert, msg, config),
-    do: colorize(:green, msg, config)
+    do: colorize(:green, show_whitespace(msg), config)
 
   defp formatter(_,  msg, _config),
     do: msg
 
+  defp show_whitespace(msg) do
+    if String.match?(msg, ~r/\A[ ]+\z/) do
```

Comment:
I considered this but thought it would be better to allow different formatters to have different behaviours here.

I could do the character substitution at the `ExUnit.Diff` level, or I could introduce a new clause for `CLIFormatter.formatter/3` for strings, but then I think we still need to do the check if the diff fragment is entirely whitespace there, unless we're OK with highlighting the background of strings when the characters are not whitespace.